### ISSUE TITLE
Add nostrcheck.me for media uploads [200.000 sats]

### ISSUE THREAD

**Issue by quentintaranpino** [damus-io/damus#786]
**New API v1 released, please contact me before start coding**

[https://github.com/quentintaranpino/nostrcheck-api-ts](https://github.com/quentintaranpino/nostrcheck-api-ts)

-------------------------------

I opened an issue a few weeks ago requesting this functionality. We have an API ready to be used to upload files.

Someone closed the issue as a duplicate of someone else.

I have been reviewing the file (AttachMediaUtility.swift) and it would not be difficult to add it.

URL with the API documentation:
https://github.com/quentintaranpino/nostrcheck-api

API's URL: 
https://nostrcheck.me/api/media.php

Basically a POST event is necessary with the following values:

<html>
<body>

Name | type
-- | --
apikey | string
publicgallery | file
type | string

</body>
</html>

Example with curl:

`curl --location 'https://nostrcheck.me/api/media.php' \
--form 'publicgallery=@"FVoK1TKy9/file.jpg"' \
--form 'apikey="26d075787d261660682fb9d20dbffa538c708b1eda921d0efa2be95fbef4910a"' \
--form 'type="media"'`

**User apikey** 
Also it would be great if you create a new field (as in the translation settings) with the possibility of specifying a user's apikey, if specified the uploads will be public anyway (they all are, they are URL's) but they will be integrated in the user's gallery.

![image](https://nostrcheck.me/media/quentin/nostrcheck.me_5521098091129817061683356881.jpg)

Example with my gallery:
https://nostrcheck.me/gallery/quentin

Other galleries:
![image](https://nostrcheck.me/media/quentin/nostrcheck.me_6654401988708114081681901718.jpg)

I think adding alternatives will help **decentralize the network**.😀

**Comment by lima46zulu** [comment]
I agree. It's unfortunate that file sharing on Nostr is so centralized; adding more options for file uploads will help, at least partially.

**Comment by alltheseas** [comment]
Can you add bounty amount in [xxx sats] to title @quentintaranpino ?

**Comment by quentintaranpino** [comment]
Yes!

**Comment by quentintaranpino** [comment]
I close this issue. definitely nobody intends to integrate it and I don't have a development environment on Mac OS to be able to do it myself.

**Comment by alltheseas** [comment]
@quentintaranpino there is a backlog of a few hundred things to work on. This does not mean that no one will work on this. 

Current focus is elsewhere: 
1) figuring out how to appease apple as not to get banned from the appstore by changing the zap flow
2) fixing navigation bugs
3) fixing relay disconnect bugs
4) updating design 

Btw, no one, or very few people knew Swift/iOS development before starting on Damus. I recommend keep it open, ask around for iOS devs not contributing to Damus, borrow someone's MacOS machine for the weekend. 

I appreciate the work you do, and would like to have content hoster options on Damus.

**Comment by quentintaranpino** [comment]
You are very kind @alltheseas,  I reopen the issue but I must update the data because now we have a more powerful API v1 :)

**Comment by alltheseas** [comment]
> You are very kind @alltheseas, I reopen the issue but I must update the data because now we have a more powerful API v1 :)

Sounds good @quentintaranpino - let me know when you have updated with latest info 🙏 with a comment in this thread. Added this to roadmap for devs to pick up.

**Comment by unwired01** [comment]
@quentintaranpino 

Which API should we use?

https://github.com/quentintaranpino/nostrcheck-api-ts
https://github.com/quentintaranpino/nostrcheck-api

**Comment by quentintaranpino** [comment]
Hello @unwired01 the first one:

https://github.com/quentintaranpino/nostrcheck-api-ts

Contact me on nostr or telegram if you have any questions

https://nostrcheck.me/u/quentin
https://t.me/quentintaranpino

You need NIP98 and this two fields, 

![image](https://github.com/damus-io/damus/assets/125748180/be42c826-9982-4346-8e42-998e47d2e0f2)

**If the "uploadtype" is not specified the server interpret is as normal media (not an avatar or banner).**

**Comment by quentintaranpino** [comment]
Also you have to use this other endpoint to get the status and information about a file

![image](https://github.com/damus-io/damus/assets/125748180/1d3308a7-5aff-4be5-94fe-68ae83ca4cea)

**Comment by unwired01** [comment]
Why do I need to use the "GET" method?
Why does the "POST" response not return a url?
It can save time for uploading.

**Comment by quentintaranpino** [comment]
> Why do I need to use the "GET" method?
> 
> Why does the "POST" response not return a url?
> 
> It can save time for uploading. 
> 
> 

The post returns this fields, also de URL of course.

![image](https://github.com/damus-io/damus/assets/125748180/579e9d09-75be-4dec-a2c3-6cff5772ca01)

But on the server side we transform the files and this transformation is made by an asynchronous queue, you need the get method to know the status of this transformation.

**Comment by unwired01** [comment]
I believe Damus will be impacted in some way. 
In particular for the progress bar
@jb55 what do you think about making GET request after uploading media (POST request)?

**Comment by quentintaranpino** [comment]
Hello.

I close this request, if in the future you want to integrate with the services offered by nostrcheck I will be delighted, the project will focus entirely on self-hosting for users and I will channel my efforts to promote NIP-96, first implementing it and then encouraging its use. 

[https://github.com/arthurfranca/nips/blob/nip-95-contender/96.md](https://github.com/arthurfranca/nips/blob/nip-95-contender/96.md)

This way nobody else will have to ask nostr clients developers to add their server or services, it will be a decision to be taken by the users.

Best regards.
